rclpy/rclpy/names_and_types.py

```diff
@@ -29,12 +29,14 @@ def __init__(self, **kwargs):
         self.topic_names = kwargs.get('topic_names', list())
         self.type_names = kwargs.get('type_names', list())
 
-    def __repr__(self):
-        s = 'number of topics = ' + repr(self._topic_count) + '\n'
-        s += '[topic_name] : [ROS type name]\n'
+    def __str__(self):
+        s = ''
         for i in range(self._topic_count):
-            s += '{} : {}\n'.format(
+            s += '{} [{}]'.format(
                 self._topic_names[i], self._type_names[i])
+            if i == self._topic_count - 1:
+                break
+            s += '\n'
```

Comment:
Instead of adding the newline based on the index a common approach is this:

```
lines = []
for i in range(self._topic_count):
    lines.append('{} [{}]'.format(self._topic_names[i], self._type_names[i]))
return '\n'.join(lines)
```